R/commas_linter.R

```diff
@@ -58,7 +71,11 @@ commas_linter <- function() {
     @line1 = preceding-sibling::*[1]/@line1 and
     not(preceding-sibling::*[1][self::OP-COMMA or self::EQ_SUB])
   ]"
-  xpath_after <- "//OP-COMMA[@line1 = following-sibling::*[1]/@line1 and @col1 = following-sibling::*[1]/@col1 - 1]"
+  xpath_after <- paste0(
+    "//OP-COMMA[@line1 = following-sibling::*[1]/@line1 and @col1 = following-sibling::*[1]/@col1 - 1",
+    if (allow_trailing_comma) " and not(following-sibling::*[1]/self::OP-RIGHT-BRACKET)",
```

Comment:
Maybe also allow `RBB` (`a[[1,]]`) and `OP-RIGHT-PAREN` (`a(1,)`) if the argument is set?